templates/admin/src/components/theme/list/ListSortMenu.tsx

```diff
@@ -0,0 +1,46 @@
+import { Icon, MenuItem, Select, SelectChangeEvent } from '@mui/material';
+import { getGlobalInstance } from 'plume-ts-di';
+import React from 'react';
+import MessageService from '../../../i18n/messages/MessageService';
+import { ListSortMenuProps } from '../../../lib/plume-admin-theme/list/ListProps';
+
+function ListSortMenu({ sortedObjectKey, sortPossibilities, onSort, defaultSortPossibility }: ListSortMenuProps) {
+  const messages = getGlobalInstance(MessageService).t();
+
+  const handleSortingBar = (event: SelectChangeEvent<string>) => {
+    const sortChoice = sortPossibilities.filter(
+      element => element.sortKey.toLowerCase()
+        .localeCompare(event.target.value.toString().toLowerCase())
+    )[0];
+    if (!sortChoice) {
+      return;
+    }
+    onSort(sortChoice);
+  };
+
+  return (
+    <div className="list-sort-menu">
+      <Icon>sort</Icon>
+      <Select
+        className="sort-menu"
+        defaultValue={defaultSortPossibility.sortKey}
+        onChange={(event: SelectChangeEvent<string>) => handleSortingBar(event)}
+        IconComponent={() => <Icon>expand_more</Icon>}
+      >
+        {
+          sortPossibilities.map((sortElement) => (
+            <MenuItem
+              key={sortElement.sortKey}
+              value={sortElement.sortKey}
+            >
+              {(messages['sort'] as any)[sortedObjectKey][sortElement.sortKey.toLowerCase()]}
```

Comment:
tu peux tenter de typer _sort_ dans le Translation.ts comme étant un objet contenant que des clés du type A et typer _sortedObjectKey_ comme étant une clé du type A. Il y a peut être un truc à jouer avec la propriété "in" en ts.
ex : `type: { [key in EngagementType] : string },`